### PR TITLE
k3s_1_26: 1.26.4+k3s1 -> 1.26.5+k3s1

### PR DESCRIPTION
**Pull request by euank**
###### Description of changes

This updates k3s_1_26 to the latest version.

Upstream release notes here: https://github.com/k3s-io/k3s/releases/tag/v1.26.5%2Bk3s1

This also, notably, updates the "update.sh" script to work for both 1.26 and 1.27, so hopefully future updates happen a bit more automatically.

Effectively, this plumbs in a minor version and uses grep to filter down tags when looking for updates.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests)
  - `nix build '.#k3s_1_26.tests.multi-node'`
  - `nix build '.#k3s_1_26.tests.single-node'`
  - `nix build '.#k3s_1_27.tests.multi-node'`
  - `nix build '.#k3s_1_27.tests.single-node'`
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/k3s/update-script.sh [hunk 1]
```diff
@@ -3,19 +3,24 @@
 
 set -x -eu -o pipefail
 
+MINOR_VERSION="${1:?Must provide a minor version number, like '26', as the only argument}"
+
 WORKDIR=$(mktemp -d)
-trap "rm -rf ${WORKDIR}" EXIT
+# trap "rm -rf ${WORKDIR}" EXIT
```
Comment (yajo) [inline pkgs/applications/networking/cluster/k3s/update-script.sh:9]: This seems to have been forgotten here?

**Review by yajo — APPROVED** [review]
Great, thanks!

File: pkgs/applications/networking/cluster/k3s/update-script.sh (hunk 1 above)
Comment (euank) [reply pkgs/applications/networking/cluster/k3s/update-script.sh]: Fixed, thank you!

**Comment by euank** [comment]
Since https://github.com/NixOS/nixpkgs/pull/239204 merged, I've rebased this to remove the update script change (which was duplicated there), and to also remove the `multicallContainerd` builder option, since it's not needed anymore.

Tests still pass

**Review by yajo — APPROVED** [review]

**Comment by yajo** [comment]
Good! Am I able to merge somehow? 🤔

**Review by farcaller — APPROVED** [review]

**Comment by euank** [comment]
I see that there's now a 1.26.6.

I'll run `nix-shell maintainers/scripts/update.nix --argstr package k3s_1_26`, verify tests pass on that, and squash that commit into this.

Might as well update us as far forward as we can if there's no surprises there.

**Comment by RaitoBezarius** [comment]
@euank Feel free to push a new upgrade for 1.26.6, I prefer we get those updates going :).

**Comment by euank** [comment]
Yup! I'll open a new PR for that in a sec, as soon as tests pass :)

**Comment by euank** [comment]
Since the package got restructured since we cut the release branch, this one will need a manual backport.
I'll pick that up.

That should then let the 1.26.6 be able to go through the above automatic backport process too, which'll make that PR easier to get merged too I think.

**Pull request merged**